pkg/config/autodiscovery/autodiscovery.go

```diff
@@ -30,13 +30,33 @@ func DiscoverComponentsFromConfig() ([]config.ConfigurationProviders, []config.L
 		detectedProviders = append(detectedProviders, prometheusProvider)
 	}
 
-	advancedConfigs, _, err := providers.ReadConfigFiles(providers.WithAdvancedADOnly)
-	if err != nil {
-		log.Debugf("Couldn't read config files: %w", err)
-	}
+	// Auto-add file-based kube service and endpoints config providers based on check config files.
+	if flavor.GetFlavor() == flavor.ClusterAgent {
+		advancedConfigs, _, err := providers.ReadConfigFiles(providers.WithAdvancedADOnly)
+		if err != nil {
+			log.Warnf("Couldn't read config files: %w", err)
+		}
+
+		svcFound, epFound := false, false
+		for _, conf := range advancedConfigs {
+			for _, adv := range conf.AdvancedADIdentifiers {
+				if !svcFound && !adv.KubeService.IsEmpty() {
+					svcFound = true
+					log.Info("Configs with advanced kube service identifiers detected: Adding the 'kube service file' config provider")
+					detectedProviders = append(detectedProviders, config.ConfigurationProviders{Name: names.KubeServicesFileRegisterName, Polling: false})
+				}
 
-	if len(advancedConfigs) > 0 && flavor.GetFlavor() == flavor.ClusterAgent {
-		detectedProviders = append(detectedProviders, config.ConfigurationProviders{Name: "kube_services_file", Polling: false})
+				if !epFound && !adv.KubeEndpoints.IsEmpty() {
+					epFound = true
+					log.Info("Configs with advanced kube endpoints identifiers detected: Adding the 'kube endpoints file' config provider")
+					detectedProviders = append(detectedProviders, config.ConfigurationProviders{Name: names.KubeEndpointsFileRegisterName, Polling: true})
```

Comment:
could you add a comment that explain why `Polling: true` for `endpoints` but not for `service`

Reply:
done